app/src/main/java/eu/kanade/presentation/more/settings/screen/SettingsSecurityScreen.kt

```diff
@@ -96,6 +118,74 @@ object SettingsSecurityScreen : SearchableSettings {
+                        onReturnPassword = { password ->
+                            dialogOpen = false
+
+                            CbzCrypto.deleteKey()
+                            securityPreferences.cbzPassword().set(CbzCrypto.encrypt(password.replace("\n", "")))
+                        },
+                    )
+                }
+                Preference.PreferenceItem.TextPreference(
+                    title = stringResource(R.string.set_cbz_zip_password),
+                    onClick = {
+                        dialogOpen = true
+                    },
+                )
+            },
+            Preference.PreferenceItem.TextPreference(
+                title = stringResource(R.string.delete_cbz_archive_password),
+                onClick = {
+                    CbzCrypto.deleteKey()
+                    securityPreferences.cbzPassword().set("")
+                    context.toast(R.string.password_successfully_deleted, Toast.LENGTH_SHORT).show()
+                },
+                enabled = CbzCrypto.isPasswordSet(),
+            ),
+            Preference.PreferenceItem.ListPreference(
+                pref = securityPreferences.localCoverLocation(),
+                title = stringResource(R.string.save_local_manga_covers),
+                entries = SecurityPreferences.CoverCacheLocation.values()
+                    .associateWith { stringResource(it.titleResId) },
+                enabled = securityPreferences.passwordProtectDownloads().get(),
+                onValueChanged = {
+                    try {
+                        CbzCrypto.deleteLocalCoverCache(context)
+                        CbzCrypto.deleteLocalCoverSystemFiles(context)
+                        true
+                    } catch (e: Exception) {
+                        logcat(LogPriority.ERROR, e)
+                        context.toast(e.toString(), Toast.LENGTH_SHORT).show()
+                        false
+                    }
+                },
+            ),
+            Preference.PreferenceItem.TextPreference(
+                title = stringResource(R.string.delete_cached_local_source_covers),
+                subtitle = stringResource(R.string.delete_cached_local_source_covers_subtitle),
+                onClick = {
+                    try {
+                        CbzCrypto.deleteLocalCoverCache(context)
+                        CbzCrypto.deleteLocalCoverSystemFiles(context)
+                        context.toast(R.string.successfully_deleted_all_locally_cached_covers, Toast.LENGTH_SHORT).show()
+                    } catch (e: Exception) {
+                        logcat(LogPriority.ERROR, e)
+                        context.toast(R.string.something_went_wrong_deleting_your_cover_images, Toast.LENGTH_LONG).show()
+                    }
+                },
+                enabled = context.getExternalFilesDir("covers/local")?.absolutePath?.let { File(it).listFiles()?.isNotEmpty() } == true,
```

Comment:
This should be done in a background thread, this code may work but I didn't test it directly.
```suggestion
enabled = produceState(false) {
    withIOContext {
        value = context.getExternalFilesDir("covers/local")?.absolutePath?.let { File(it).listFiles()?.isNotEmpty() } == true
    }
}.value
```